swaylock/render.c

```diff
@@ -0,0 +1,146 @@
+#define _POSIX_C_SOURCE 199506L
+#include <math.h>
+#include <stdlib.h>
+#include <wayland-client.h>
+#include "cairo.h"
+#include "background-image.h"
+#include "swaylock/swaylock.h"
+
+#define M_PI 3.14159265358979323846
+
+void render_frame(struct swaylock_surface *surface) {
+	struct swaylock_state *state = surface->state;
+	surface->current_buffer = get_next_buffer(state->shm,
+			surface->buffers,
+			surface->width * surface->scale,
```

Comment:
Define `buffer_width` before this instruction, and replace `surface->width * surface->scale` with `buffer_width`